### ISSUE TITLE
Create a snap package

### ISSUE THREAD

**Issue by nm17** [tenacityteam/tenacity-legacy#279]
- [x] I have read the specified guidelines for issues

**Is your feature request related to a problem? Please describe.**
Since Audacity has a snap package, it would be great if Tenacity would be ready for one when the first release happens.
Snap is really wide spread, so I think we should have one.

**Describe the solution you'd like**
Currently, I'm working on my own version at https://github.com/nm17/tenacity-snap . Any help with it would be appreciated!

**Describe alternatives you've considered**
Other distributions are already being worked on.

**Additional context**
I've already consulted people on the IRC, so I didn't just start it out of the blue. Any help with this would be greatly appreciated.  

Currently I need help with https://github.com/nm17/tenacity-snap/issues/1

**Comment by fossdd** [comment]
Really thanks for your work. Maybe the admins (@tenacityteam/maintainers) can also mirgrate that git repo to our GitHub orga and to sourcehut if the bug is fixed and all works fine. We could publish snap packages to snapcraft.io automatically via CI?

**Comment by nbsp** [comment]
imo, I'd rather the Snap package stay unofficial and community-maintained. I'm against Canonical's attempt to use proprietary servers to create a "distro-independent" package manager.

**Comment by n0toose** [comment]
I guess this will be a part of our "infrastructure", so if they (@nm17) want our blessing or to use the Tenacity name instead of their own, I don't see a reason for all of us to put a stop to it and shoot ourselves in the foot?

**Comment by nbsp** [comment]
I'm fine with using the "Tenacity" name, I was referring to official support from the team here and on IRC. Many projects have unofficial ports to various platforms

**Comment by eli-schwartz** [comment]
Snap is a lot less widespread than flatpak, since the former is FOSS-client-proprietary-server and few really buy into it other than Canonical, while flatpak is community driven and fully FOSS.

It's probably a better idea to look into flatpak.

Edit: there is already #25, I doubt more than one distro independent packaging format is needed.

**Comment by CHJ85** [comment]
For starters. Let's wait till the rebrand is done before even thinking about packaging to either Snap, Flatpak or AppImage for that matter.

**Comment by cpcbegin** [comment]
I prefer AppImage because it's more clean.
Anyway, this new packages MUST BE an option, better keep tradicional packages.

**Comment by dCo3lh0** [comment]
+1 for the AppImage, its easier and works.

**Comment by processor286** [comment]
I'd be comfortable with appimage or flatpack. I thing snap has a place where you want the maintainer to push updates to perhaps security related issues (e.g. maybe you run the Snap of NextCloud and would like it to auto-update).  For desktop applications though, appimage or flatpack is ideal.

I'm very much reminded of xkcd of course when I think of this.  We have 7 packaging formats, what we need is one to replace them all.  Now we have 8 formats.

**Comment by Lvaskz** [comment]
> +1 for the AppImage, its easier and works.

I agree,see here https://github.com/AppImageCrafters/appimage-builder
much easier now

**Comment by fossdd** [comment]
> > +1 for the AppImage, its easier and works.
> 
> I agree,see here https://github.com/AppImageCrafters/appimage-builder
> much easier now

The "nighly" (for every commit) builds already built AppImages, this shouldn't be a problem.

This can also be part of https://github.com/tenacityteam/tenacity/issues/166 (if we keep github releases as binary platform).

**Comment by TheEvilSkeleton** [comment]
I'm completely against snap, even if the backend was FOSS. snaps were designed for servers, not for applications. It compresses the application which makes it take a couple of seconds just to launch due to decompression, which varies on the size of the software, and GTK theming is awful at best.

Don't use software on places that weren't supposed to be used. Using snaps for apps is like using Flatpak on servers.

**Comment by sickcodes** [comment]
Snap is just the Canonical App Store.

What's wrong with just a debian package?

**Comment by processor286** [comment]
> What's wrong with just a debian package?

Nothing [apart from dependency difficulties] if you just want to run on Debian or derivatives.

I don't like snap for desktop packages, but flatpak or appimage would be perfect.

Personally I run a mix of Arch, OpenSuSE, and Slackware so a .deb wouldn't be much use for me.

Linux dependencies are hard, and that's the problem that flat/appimage and even snap are trying to solve [ among other things ].

**Comment by ghost** [comment]
.appimage, please.
/EDIT:
Here are unstable builds including AppImage (described as "Ubuntu"):
https://nightly.link/tenacityteam/tenacity/workflows/cmake_build/master

**Comment by nbsp** [comment]
An AppImage is already generated by the Ubuntu build. Please don't clutter this issue with your gripes about Snap.

**Comment by CHJ85** [comment]
+1 to @SFR-git. Besides, AppImage is easier to build and maintain.
So I think (for now at least), let's just stick with this.
